trino/Dockerfile

```diff
@@ -4,56 +4,62 @@
 # Ignoring DL4006 globally because we inherit the SHELL from our base image
 # hadolint global ignore=DL3038,DL4006
 
-FROM stackable/image/java-base
+FROM stackable/image/java-devel AS storage-connector-builder
 
-ARG PRODUCT
-# External OPA authorizer only required for 414 and 428, included in 438
-# The `OPA_AUTHORIZER` is not set in the conf.py for version 438 to remove the ARG as soon as 414 and 428 are removed
-ARG OPA_AUTHORIZER
-ARG JMX_EXPORTER
-ARG RELEASE
 ARG STORAGE_CONNECTOR
+ARG STORAGE_CONNECTOR_GIT_REV
 
-LABEL name="Trino" \
-      maintainer="info@stackable.tech" \
-      vendor="Stackable GmbH" \
-      version="${PRODUCT}" \
-      release="${RELEASE}" \
-      summary="The Stackable image for Trino." \
-      description="This image is deployed by the Stackable Operator for Trino."
+WORKDIR /stackable
 
-RUN microdnf update && \
-    microdnf install \
-    gzip \
-    httpd-tools \
-    python \
-    tar \
-    zip && \
-    microdnf clean all && \
-    rm -rf /var/cache/yum
+RUN git clone https://github.com/snowlift/trino-storage --branch=v${STORAGE_CONNECTOR} --depth=1 && \
+    cd trino-storage && git checkout ${STORAGE_CONNECTOR_GIT_REV}
+RUN --mount=type=cache,target=/root/.m2/repository cd trino-storage && \
+    # Upstream builds are marked as -SNAPSHOT, even for release builds
+    mvn versions:set -DnewVersion=${STORAGE_CONNECTOR} && \
+    mvn package -DskipTests
+
+# # TODO: remove the OPA authorizer builder once Trino versions 414 and 428 are removed
+FROM stackable/image/java-devel AS opa-authorizer-builder
```

Comment:
```suggestion
# Not tagging base image because it is built as part of the same process
# hadolint ignore=DL3006
FROM stackable/image/java-devel AS opa-authorizer-builder
```